### ISSUE TITLE
Poor CP Performance

### ISSUE THREAD

**Issue by superflausch** [supercool/Pimp-My-Matrix#30]
Hi there

Recently I noticed more and more laggy entry edit pages (CP). Scrolling is laggy and editing fields is often delayed and not very responsive. The browser seems to not be able to handle the rendering of the page. We use a lot of matrix blocks on our entries and added some tabs with Pimp-My-Matrix. It feels like some JS script is dragging down the browser.

Not sure if this is a Pimp-My-Matrix specific issue, but deactivating the plugin helps to improve the situation. Still not optimal but usable. Maybe it's also a combination of several plugins and the big amount of fields leading to the issue.

Is this maybe something you are aware of? What info can I provide you to help resolve and improve the situation? 

I love the plugin and especially the new release so keep up the good work. I really hope I can keep using the plugin.

Cheers.

Pascal

**Comment by joshangell** [comment]
Thanks for the report Pascal, I'll take a look into it in the new year at some point. It is highly likely due to the number of blocks and block types you have and the fact that I'm listening to a couple of element resize events.

**Comment by superflausch** [comment]
Yeah I have a lot of blocks, also a lot of block-types and they tend to have a couple of fields. I had to deactivate the plugin for now.

Please let me know if I can help. I really like the functionality the plugin adds.

**Comment by richhayler** [comment]
Hi Josh,
Don't know if you've received much more feedback on the CP performance issues but from my perspective this plugin has become completely unusable - at least in the most recent CMS update. Performance is so laggy (constantly beachballs so have to kill the browser) when attempting to add or edit with the pimp my matrix config modals. In my opinion this plugin filled a big gap by allowing more control over the layout of a complex entry page and has been a default add on to any new project in the past. Would be great to have this issue sorted out soon to take advantage of the fantastic features your plugin brings to making the CP more usable.

**Comment by supercooljames** [comment]
Hi Rich

Thanks for the feedback and sorry to hear you're having issues. Just to give you a quick response on this – We've noticed slight performance issues on one or two projects with a lot of blocks (nothing too crippling though at this end), so we're going to need to address it.

We're currently having to prioritise another project, but aim to pick this up in a month or so.

**Comment by mjatharvest** [comment]
I've noticed Safari seems to struggle more than Firefox (on OS X 10.11.2). In your experience, are the performance issues worse in certain browsers?

**Comment by richhayler** [comment]
Yes, in Safari I have to close the tab due to the unresponsiveness of the PMM interface. In Chrome and Firefox it seems to work fine.
